File: generated_code/usr/Composition_vp/5b441e211d41c814fc6ceaf1/Swift/class2.swift
class class2 {

// attributes: {"class2attribute1"}
var tableName = "class2"

func createOne(data: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        adapter.createOne(tableName: tableName, data: data, successCB: successCB, errorCB: errorCB)
}

func createMany(data: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        adapter.createMany(tableName: tableName, data: data, successCB: successCB, errorCB: errorCB)
}

func readOne(data: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        adapter.readOne(tableName: tableName, data: data, successCB: successCB, errorCB: errorCB)
}

func readMany(data: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        adapter.readMany(tableName: tableName, data: data, successCB: successCB, errorCB: errorCB)
}

func update(search: JSON, update: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        let data = ["oldData": search, "newData": update]
        adapter.update(tableName: collection, data: update, successCB: successCB, errorCB: errorCB)
}

func delete(data: JSON) {
        func successCB(_ result: Any)-> Void {
            print(result)
        }
        func errorCB(_ result:Any) -> Void {
            print(result)
        }
        adapter.delete(tableName: tableName, data: data, successCB: successCB, errorCB: errorCB)
}



}

File: generated_code/default/Generalization/Java/class2.java
import com.google.gson.JsonObject;


public class class2 {

    public static String className = "class2";
    public static String[] attributes = {
        "class2Attribute2", "class2Attribute1"
    };

    public static void createOne(JsonObject data) {
        class createOneCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        createOneCB CBModel = new createOneCB();
        Adapter.createOne(className, data, CBModel);
    }
    
    public static void createMany(JsonArray data) {
        class createManyCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        createManyCB CBModel = new createManyCB();
        Adapter.createMany(className, data, CBModel);
    }
    
    public static void readOne(JsonObject data) {
        class readOneCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        readOneCB CBModel = new readOneCB();
        Adapter.readOne(className, data, CBModel);
    }
    
    public static void readMany(JsonObject data) {
        class readManyCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        readManyCB CBModel = new readManyCB();
        Adapter.readMany(className, data, CBModel);
    }
    
    public static void update(JsonObject searchData, JsonObject updateData) {
        class updateCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        JsonObject data = new JsonObject();
        data.add("oldData",search);
        data.add("newData",update);
        updateCB CBModel = new updateCB();
        Adapter.update(className, data, CBModel);
    }
    
    public static void delete(JsonObject data) {
        class deleteCB {
            public void successCB(String result) {
                System.out.println("successCB: " + result);
            }
            public void errorCB(String message) {
                System.out.println("errorCB: " + message);
            }
        }
        deleteCB CBModel = new deleteCB();
        Adapter.delete(className, data, CBModel);
    }
    


}
